tests/cli/cli.spec.ts

```diff
@@ -0,0 +1,103 @@
+import { expect } from "chai";
+import { execFile, fork } from "child_process";
+import { version } from "df/core/version";
+import { suite, test } from "df/testing";
+import * as fs from "fs";
+import * as path from "path";
+
+suite(__filename, () => {
+  const nodePath = "external/nodejs_linux_amd64/bin/node";
+  const npmPath = "external/nodejs_linux_amd64/bin/npm";
+  const corePackageTarPath = "packages/@dataform/core/package.tgz";
+  const cliEntryPointPath = "tests/cli/node_modules/@dataform/cli/bundle.js";
+
+  const projectDir = path.join(process.env.TEST_TMPDIR, "project");
+  const npmCacheDir = path.join(process.env.TEST_TMPDIR, "npm-cache");
+
+  test("init and compile", async () => {
+    // Initialize a project using the CLI, don't install packages.
+    const initChildProcess = fork(cliEntryPointPath, [
+      "init",
+      "redshift",
+      projectDir,
+      "--skip-install"
+    ]);
+
+    await new Promise(resolve => {
+      initChildProcess.on("close", resolve);
+    });
+
+    // Install packages manually to get around bazel sandbox issues.
+    const installChildProcess = execFile(npmPath, [
+      "install",
+      "--prefix",
+      projectDir,
+      "--cache",
+      npmCacheDir,
+      corePackageTarPath
+    ]);
+    installChildProcess.stdout.pipe(process.stdout);
+    installChildProcess.stderr.pipe(process.stderr);
+
+    await new Promise(resolve => {
+      installChildProcess.on("close", resolve);
+    });
+
+    // Write a simple file to the project.
+    fs.writeFileSync(
+      path.join(projectDir, "definitions", "example.sqlx"),
+      `
+config { type: "table" }
+select 1 as test
+`
+    );
+
+    // Compile the project using the CLI.
+    const compileChildProcess = execFile(nodePath, [
+      cliEntryPointPath,
+      "compile",
+      projectDir,
+      "--json"
+    ]);
+    compileChildProcess.stdout.pipe(process.stdout);
```

Comment:
nit: i would suggest naming these according to the thing that is being run, not necessarily the command. e.g. `cliChildProcess` and `npmChildProcess` (above). IMO it makes it clearer what is being run, and avoids confusion (i thought for a few seconds that this meant the *actual* compile child process, but it doesn't of course).

Reply:
Kind of not relevant now, have refactored